### PR TITLE
Build images with alpine to resolve vulnerabilities

### PR DESCRIPTION
**Pull request by peterth3**
## What
*Describe what the change is solving*
Upgrades dependencies for several docker images and switches to alpine to improve image footprint and security.

## How
*Describe the solution*

Switches base images to alpine wherever possible and changes dependency installs from `yum` to `apk`.

## Recommended reading order
n/a

## Can this PR be safely reverted / rolled back?
*If you know that your PR is backwards-compatible and can be simply reverted or rolled back, check the YES box.*

*Otherwise if your PR has a breaking change, like a database migration for example, check the NO box.*

*If unsure, leave it blank.*
- [x] YES 💚
- [ ] NO ❌

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

Not to our knowledge. We are running these alpine-based images in our cluster without issues.

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks for the contribution @peterth3 I requested to our engineer team to take a look in your contribution.

File: airbyte-workers/build.gradle [hunk 1]
```diff
@@ -15,7 +15,7 @@ configurations.all {
     resolutionStrategy {
         // Ensure that the versions defined in deps.toml are used
         // instead of versions from transitive dependencies
-        force libs.flyway.core, libs.jooq, libs.s3, libs.aws.java.sdk.s3
+        force libs.flyway.core, libs.jooq, libs.s3, libs.aws.java.sdk.s3, libs.jsonsmart
```
Comment (marcosmarxm) [inline airbyte-workers/build.gradle:18]: Any particular reason to add `jsonsmart` lib?

File: airbyte-webapp/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,17 @@
 ARG NGINX_IMAGE=nginx:alpine
 FROM ${NGINX_IMAGE}
 
+RUN apk update && apk upgrade && apk upgrade -U openssl
 EXPOSE 80
 
+#RUN addgroup -S appgroup && \
+#    adduser -S appuser -G appgroup -u 1000 \
```
Comment (marcosmarxm) [inline airbyte-webapp/Dockerfile:8]: Can you remove the comments?

File: airbyte-container-orchestrator/Dockerfile [hunk 3]
```diff
@@ -3,8 +3,9 @@ FROM ${JDK_IMAGE} AS orchestrator
 
 ARG DOCKER_BUILD_ARCH=amd64
 
-RUN amazon-linux-extras install -y docker
-RUN yum install -y jq tar && yum clean all
+#RUN amazon-linux-extras install -y docker
+#RUN yum install -y jq tar && yum clean all
```
Comment (marcosmarxm) [inline airbyte-container-orchestrator/Dockerfile:7]: Remove the comments.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
I am doing the first review, @peterth3, but we must wait for the engineering team's review and their considerations.

File: airbyte-workers/build.gradle (hunk 1 above)
Comment (mauricioalarcon) [reply airbyte-workers/build.gradle:18]: yes, the provided version has know vulnerabilities, so we bumped it and force it here. 

https://nvd.nist.gov/vuln/detail/CVE-2023-1370

**Comment by davinchia** [comment]
the security changes here look sane to me, @wennergr can you confirm?

**Comment by davidmhleung** [comment]
@davinchia , @wennergr - Any update on this?

**Comment by davidmhleung** [comment]
@marcosmarxm - requested changes have been updated.

**Comment by marcosmarxm** [comment]
> @marcosmarxm - requested changes have been updated.

I'm going to take a look tomorrow @davidmhleung

**Comment by peterth3** [comment]
Thank you @marcosmarxm 

We need to resolve image vulnerabilities to run Airbyte in our environment. Here's the latest security scan on ArtifactHub ("F"), which aligns with the results we're seeing from our internal scanner. Switching to alpine and upgrading auto-temporal resolved the majority of these vulnerabilities on our end.

https://artifacthub.io/packages/helm/airbyte/airbyte
Screenshot 2023-06-02 at 9 44 49 AM

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
> Thank you @marcosmarxm
> 
> We need to resolve image vulnerabilities to run Airbyte in our environment. Here's the latest security scan on ArtifactHub ("F"), which aligns with the results we're seeing from our internal scanner. Switching to alpine and upgrading auto-temporal resolved the majority of these vulnerabilities on our end.
> 
> https://artifacthub.io/packages/helm/airbyte/airbyte Screenshot 2023-06-02 at 9 44 49 AM

Understood. But today uprading temporal won't work without doing changes to Airbyte code first to make compatible.
Can we proceeded as described in the email: upgrade the packages and addition of `jsonsmart` library.

**Pull request merged**